src/main/java/com/amazonaws/encryptionsdk/keyrings/KmsKeyring.java

```diff
@@ -77,15 +80,15 @@ public void onEncrypt(EncryptionMaterials encryptionMaterials) {
 
         // If the input encryption materials do not contain a plaintext data key and this keyring does not
         // have a generator defined, OnEncrypt MUST not modify the encryption materials and MUST fail.
-        if (!encryptionMaterials.hasPlaintextDataKey() && generatorKeyId == null) {
+        if (encryptionMaterials.getCleartextDataKey() == null && generatorKeyId == null) {
             throw new AwsCryptoException("Encryption materials must contain either a plaintext data key or a generator");
         }
 
         final List<String> keyIdsToEncrypt = new ArrayList<>(keyIds);
 
         // If the input encryption materials do not contain a plaintext data key and a generator is defined onEncrypt
         // MUST attempt to generate a new plaintext data key and encrypt that data key by calling KMS GenerateDataKey.
-        if (!encryptionMaterials.hasPlaintextDataKey()) {
```

Comment:
While not a major issue, I like having `hasPlaintextDataKey` as it means we can check for presence without dropping more copies of a cryptographic key throughout our memory.

Reply:
Yeah I like it too, it was more necessary in the other materials classes since I was throwing an InvalidStateException if the plaintext data key was null. I removed it since it wasn't strictly necessary anymore, but it does improve readability somewhat. I'll add it back